packages/hash/frontend/src/components/BlockLoader/shared.ts

```diff
@@ -0,0 +1,112 @@
+import { JsonObject } from "@blockprotocol/core";
+import { BlockGraph, Entity } from "@blockprotocol/graph";
+import get from "lodash/get";
+import set from "lodash/set";
+
+export type SchemaMap = {
+  mapId: string;
+  transformations?: {
+    [destinationKey: string]:
+      | {
+          default?: any;
+          sourceKey?: string;
+        }
+      | undefined;
+  };
+};
+
+type JsonPath = `$.${string}`;
+
+const isJsonPath = (path: unknown): path is JsonPath =>
+  typeof path === "string" && path.startsWith("$.");
+
+/** strips the leading "$." from a jsonpath, if present */
+const pathWithoutLeadingChars = (jsonPath: JsonPath | string) =>
+  jsonPath.replace(/^\$\./, "");
+
+/**
+ * Converts a source entity, linkedEntities and linkGroups into a data tree
+ * @todo support linkedAggregations
+ */
+export const dataTreeFromEntityGraph = (
+  entity: Pick<Entity, "entityId" | "properties">,
+  graphFromEntity: BlockGraph,
+  processedEntityProperties: Map<string, JsonObject> = new Map(),
+) => {
+  const entityPropertiesClone = JSON.parse(JSON.stringify(entity.properties));
+  const linkGroups = graphFromEntity.linkGroups.filter(
+    ({ sourceEntityId }) => entity.entityId === sourceEntityId,
+  );
+  for (const linkGroup of linkGroups) {
+    if (!isJsonPath(linkGroup.path)) {
+      throw new Error(
+        `Expected linkGroup.path to be jsonpath, got: ${linkGroup.path}`,
+      );
+    }
+    const entitiesOnPath = [];
```

Comment:
I think this is implicitly `any` typed

Reply:
Typed in https://github.com/hashintel/hash/pull/918/commits/b4eae05c32e1e29eaf0078b90e790874984398da